File: code/java/src/main/java/abstractfactory/ParkVisitor.java
package abstractfactory;

class ParkVisitor {
    public static void main(String[] args) {
        // Choose themed area (factory)
        AbstractFoodStallFactory adventureLandFactory = new AdventureLandFactory();
        AbstractFoodStallFactory fantasyLandFactory = new FantasyLandFactory();

        // Order themed food items
        IceCream tropicalIceCream = adventureLandFactory.createIceCream();
        Pizza exoticPizza = adventureLandFactory.createPizza();

        IceCream magicalIceCream = fantasyLandFactory.createIceCream();
        Pizza enchantedPizza = fantasyLandFactory.createPizza();

        // Enjoy themed food items
        tropicalIceCream.enjoy();
        exoticPizza.enjoy();
        magicalIceCream.enjoy();
        enchantedPizza.enjoy();
    }
}

File: code/typescript/src/abstractFactory/ParkVisitor.ts
import { AbstractFoodStallFactory } from './AbstractFoodStallFactory';
import { IceCream } from './IceCream';
import { Pizza } from './Pizza';
import { AdventureLandFactory } from './AdventureLandFactory';
import { FantasyLandFactory } from './FantasyLandFactory';

// Choose themed area (factory)
const adventureLandFactory: AbstractFoodStallFactory = new AdventureLandFactory();
const fantasyLandFactory: AbstractFoodStallFactory = new FantasyLandFactory();

// Order themed food items
const tropicalIceCream: IceCream = adventureLandFactory.createIceCream();
const exoticPizza: Pizza = adventureLandFactory.createPizza();

const magicalIceCream: IceCream = fantasyLandFactory.createIceCream();
const enchantedPizza: Pizza = fantasyLandFactory.createPizza();

// Enjoy themed food items
tropicalIceCream.enjoy();
exoticPizza.enjoy();
magicalIceCream.enjoy();
enchantedPizza.enjoy();
